pcap-linux.c

```diff
@@ -3744,19 +3203,11 @@ prepare_tpacket_socket(pcap_t *handle)
 			/*
 			 * Success.
 			 */
-			return 1;
-		}
-		if (ret == -1) {
-			/*
-			 * We failed for some reason other than "the
-			 * kernel doesn't support TPACKET_V3".
-			 */
-			return -1;
+			return 0;
```

Comment:
Again, as per the above, I forgot that you have to distinguish between "TPACKET_V3 works", "TPACKET_V3 doesn't work because we only support TPACKET_V3", and "TPACKET_V3 doesn't work for some other reason that probably means TPACKET_V2 will fail as well", so this should treat 0 as success, -1 as a hard error, and 1 as "OK, fall through to the TPACKET_V2 code".